mlflow/tracking/client.py

```diff
@@ -910,6 +914,50 @@ def log_artifacts(self, run_id, local_dir, artifact_path=None):
         """
         self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
 
+    @contextlib.contextmanager
+    def _log_artifact_helper(self, run_id, artifact_file):
+        """
+        Yields a temporary path to store a file, and then calls `log_artifact` against that path.
+        """
+        norm_path = posixpath.normpath(artifact_file)
+        filename = posixpath.basename(norm_path)
+        artifact_path = posixpath.dirname(norm_path)
+        artifact_path = None if artifact_path == "" else artifact_path
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, filename)
+            yield tmp_path
+            self.log_artifact(run_id, tmp_path, artifact_path)
+
+    def log_text(self, run_id, text, artifact_file):
+        """
+        Log a text as an artifact.
+
+        :param run_id: String ID of the run.
+        :param text: Text to log.
+        :param artifact_file: The run-relative artifact file path to which the text is saved.
+
+        ..code-block:: python
+            :caption: Example
+
+            from mlflow.tracking import MlflowClient
+
+            client = MlflowClient()
+            run = client.create_run(experiment_id="0")
+
+            # Log a text in the artifact root
+            client.log_text(run.info.run_id, "text1", "file1.txt")
+
+            # Log a text in a subdirectory
+            client.log_text(run.info.run_id, "text2", "dir/file2.txt")
+
+            # Log an HTML text
+            client.log_text(run.info.run_id, "<h1>header</h1>", "index.html")
```

Comment:
Added this example to indicate that this API accepts any text file formats (e.g. .html, .csv, .py, .sh)